stripe/src/main/java/com/stripe/android/checkout/CheckoutViewModel.kt

```diff
@@ -36,13 +40,35 @@ internal class CheckoutViewModel internal constructor(
         mutableTransition.postValue(target)
     }
 
-    fun getPaymentMethods(
-        customerId: String,
+    /**
+     * Fetch the customer's saved payment methods.
+     * The activity args are assumed to be static across the lifetime of the view model
+     * so calling this method multiple times will not result in re-fetching the payment methods.
+     *
+     * If you wish to force an update, call [updatePaymentMethods] directly.
+     */
+    fun getPaymentMethods(activity: Activity): LiveData<List<PaymentMethod>> {
+        if (paymentMethods.value == null) {
+            val args: CheckoutActivityStarter.Args? = CheckoutActivityStarter.Args.fromIntent(activity.intent)
+            if (args == null) {
+                onError(IllegalStateException("Missing activity args"))
+            } else {
+                updatePaymentMethods(
+                    args.ephemeralKey,
+                    args.customerId
+                )
+            }
+        }
+        return paymentMethods
+    }
+
+    fun updatePaymentMethods(
         ephemeralKey: String,
+        customerId: String,
         stripeAccountId: String? = this.stripeAccountId
-    ) = liveData(workContext) {
-        val result =
-            kotlin.runCatching {
+    ) {
+        CoroutineScope(workContext).launch {
```

Comment:
Can't remember if we discussed this earlier, but should we use `viewModelScope` here?

Reply:
Yeah, we need to offload this onto a background thread given how `StripeApiRepositor#getPaymentMethods` is implemented today. `viewModelScope` is bound to `Dispatchers.Main` so that would block the main thread.